src/integrators/aov.cpp

```diff
@@ -305,27 +300,101 @@ class AOVIntegrator final : public SamplingIntegrator<Float, Spectrum> {
                 case Type::ShapeIndex:
                     *aovs++ = Float(dr::reinterpret_array<UInt32>(si.shape));
                     break;
+            }
+        }
 
-                case Type::IntegratorRGBA: {
-                        std::pair<Spectrum, Mask> result_sub =
-                            m_integrators[ctr].first->sample(scene, sampler, ray, medium, aovs, active);
-                        aovs += m_integrators[ctr].second;
-                        Color3f rgb =
-                            spectrum_to_color3f(result_sub.first, ray, active);
+        return result;
+    }
 
-                        *aovs++ = rgb.r(); *aovs++ = rgb.g(); *aovs++ = rgb.b();
-                        *aovs++ = dr::select(result_sub.second, Float(1.f), Float(0.f));
+    TensorXf render(Scene *scene,
+                    Sensor *sensor,
+                    uint32_t seed,
+                    uint32_t spp,
+                    bool develop,
+                    bool evaluate) override {
+
+        std::vector<TensorXf> inner_images;
+        for (auto& integrator : m_integrators) {
+            auto image = integrator->render(scene, sensor, seed, spp, develop, evaluate);
+            inner_images.push_back(image);
+        }
 
-                        if (ctr == 0)
-                            result = result_sub;
+        TensorXf aovs_image;
+        {
+            dr::suspend_grad<Float> scope(true);
```

Comment:
I think I'm missing something... If we're disabling gradients in this scope, how can the AOV fields be "attached" (to the AD graph) in `render_forward`/`render_backward`.

Reply:
As discussed irl, while it doesn't impact render_forward/render_backward, it doesn't really need to be there so I've removed it.